frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.unit.spec.ts

```diff
@@ -0,0 +1,378 @@
+import { merge } from "icepick";
+import { colors } from "metabase/lib/colors";
+import type {
+  ChartSettings,
+  SeriesWithBreakoutValues,
+  SeriesWithoutBreakoutValues,
+} from "../../XYChart/types";
+import { getSeriesWithColors } from "./series";
+
+const settings: ChartSettings = {
+  x: {
+    type: "ordinal",
+  },
+  y: {
+    type: "linear",
+  },
+  labels: {
+    left: "Count",
+    bottom: "Date",
+  },
+};
+
+describe("getSeriesWithColors", () => {
+  it("should return an empty series given an empty series", () => {
+    const seriesWithColors = getSeriesWithColors([], getPalette({}), settings);
+
+    expect(seriesWithColors).toEqual([]);
+  });
+
+  describe("Series without breakout values", () => {
+    const series: SeriesWithoutBreakoutValues[] = [
+      {
+        name: "Count",
+        yAxisPosition: "left",
+        type: "bar",
+        data: [
+          ["Doohickey", 3976],
+          ["Gadget", 4939],
+          ["Gizmo", 4784],
+          ["Widget", 5061],
+        ],
+        seriesKey: "count",
+      },
+    ];
+
+    it("should assign colors given series", () => {
+      const seriesWithColors = getSeriesWithColors(
+        series,
+        getPalette({}),
+        settings,
+      );
+
+      const expectedSeries = [
+        {
+          name: "Count",
+          color: "#509EE3", // brand color
+          yAxisPosition: "left",
+          type: "bar",
+          data: [
+            ["Doohickey", 3976],
+            ["Gadget", 4939],
+            ["Gizmo", 4784],
+            ["Widget", 5061],
+          ],
+        },
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("should assign colors from whitelabel colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        series,
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+        settings,
+      );
+
+      const expectedSeries = [
+        {
+          name: "Count",
+          color: "#123456", // whitelabel color
+          yAxisPosition: "left",
+          type: "bar",
+          data: [
+            ["Doohickey", 3976],
+            ["Gadget", 4939],
+            ["Gizmo", 4784],
+            ["Widget", 5061],
+          ],
+        },
+      ];
+
+      expect(seriesWithColors).toEqual(expectedSeries);
+    });
+
+    it("it should assign colors from column colors", () => {
+      const seriesWithColors = getSeriesWithColors(
+        series,
+        getPalette({ brand: "#123456", summarize: "#ffffff" }),
+        merge(settings, {
+          series_settings: {
```

Comment:
When we customize a series color. See more detail in https://github.com/metabase/metabase/pull/26211/files#r1019305648